### PR TITLE
feat(firestore-send-email): support SendGrid Dynamic Templates

### LINKED ISSUE #1409: Trigger Email - SendGrid Dynamic Templates
Trigger Email: Is there a way to trigger the Dynamic Templates on SendGrid? I don't like creating a Firebase Template Collection Admin Tool because this functionality is already on SendGrid. (HTML Edit)

### PR DESCRIPTION
**Pull request by pr-Mais**
Closes #1409.

This PR adds support to using SendGrid Dynamic Templates.

The templates can be used by adding the following object to the queue data:

```js
const record = {
  to: "test-assertion@email.com",
  sendGrid: {
    templateId: "d-61eb13...", // required
    dynamicTemplateData: {} // optional
  },
};
```

### CODE REVIEW

**Comment by pr-Mais** [comment]
For SendGrid e2e tests to pass, we need to set the `SENDGRID_FROM_EMAIL` environment variable.

File: _emulator/.firebaserc [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
   "projects": {
-    "default": "demo-test"
+    "default": "dev-extensions-testing"
```
Comment (dackers86) [inline _emulator/.firebaserc:3]: Prefixed `demo-` naming ensure no live services are accidentally called. Would this be the correct naming for this type of update?

Source: https://firebase.google.com/docs/emulator-suite/connect_firestore

Comment (pr-Mais) [reply _emulator/.firebaserc:3]: This was meant for getting the API key (for e2e tests) from cloud secrets, but I'm looking into a way to revert back to the demo project and set keys as secrets locally in CI.

**Comment by maximiliancs** [comment]
The corresponding issue has been open for some time now, and as the feature is already implemented in this PR, I'm eager to see it move forward to use it in production.

Politely asking: What's the status of your review? @jauntybrain @pr-Mais

**Comment by pr-Mais** [comment]
@maximiliancs this will be included in the next version

**Review by cabljac — APPROVED** [review]
lgtm

**Comment by baparham** [comment]
Thank you and nice work @pr-Mais this is fantastic to have available now!

**Comment by pr-Mais** [comment]
@baparham you're welcome, let me know if you face any issues using it.

**Comment by cedvdb** [comment]
> let me know if you face any issues using it.

@pr-Mais 

I'm facing this `delivery.error`: `error
"TypeError: Cannot read properties of undefined (reading 'attachments')"`

It looks like it's trying to get an attachment from a template, but I'm using sendgrid: 

```ts
    await this._database.emails.doc(id).create({
      to: emailAddress,
      sendGrid: {
        templateId: content.template,
        dynamicTemplateData: content.data,
      }
    });
```

**Comment by cedvdb** [comment]
It seems adding an empty `message: {}` solves that issue

**Pull request merged**